armicontrib/armiopenmc/inputWriters.py

```diff
@@ -655,16 +655,76 @@ def _buildComponentMaterial(component):
     )
 
     componentNuclides = component.getNuclides()
-    componentNuclideDensities = component.getNuclideNumberDensities(componentNuclides)
-    totalComponentNuclideDensity = sum(componentNuclideDensities)
+    componentNuclideDensities = {}
```

Comment:
I think this one method is getting too long and we should break it up into multiple, smaller methods.

The idea is generally that shorter methods are easier to understand, and thus easier to document, test, and generally less error prone.

Thoughts?